resource/ui/ts/models/proto.ts

```diff
@@ -113,6 +155,32 @@ module Models {
             available_range_count: number;
         }
 
+        /**
+         * Status is the common interface shared by NodeStatus and StoreStatus.
+         */
+        export interface Status{
```

Comment:
Well actually, "Status" isn't going to be generated if we move to a generation model.   Along with the various static methods I've added, "Status" should probably not be in that file (as you mentioned on your earlier comment, perhaps the generated protos should go into the typings folder, or at least a separate file).

So these types should not extend from Status.